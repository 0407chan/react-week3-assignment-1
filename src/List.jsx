import Item from './Item';

export default function List({ tasks, onClickDelete }) {
  console.log(tasks.length);
  if (tasks.length === 0) {
    return (
      <p>할 일이 없어요!</p>
    );
  }

  return (
    <ol>
      {tasks.map((task) => (
        <Item key={task.id} task={task} onClickDelete={onClickDelete} />
      ))}
    </ol>
  );
}
