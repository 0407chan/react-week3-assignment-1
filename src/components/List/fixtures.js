export const tasks = [
  { id: 1, title: 'foo' },
  { id: 2, title: 'bar' },
];
