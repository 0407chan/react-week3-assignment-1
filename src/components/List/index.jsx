import Item from '../Item';
import { EMPTY_MESSAGE } from '../../constants';

export default function List({ tasks, onClickDelete }) {
  if (tasks.length === 0) {
    return (
      <p>{EMPTY_MESSAGE}</p>
    );
  }

  return (
    <ol>
      {tasks.map((task) => (
        <Item key={task.id} task={task} onClickDelete={onClickDelete} />
      ))}
    </ol>
  );
}
