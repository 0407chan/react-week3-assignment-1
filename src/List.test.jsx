import { render } from '@testing-library/react';
import List from './List';

test('목록에 할일이 없음을 알 수 있다.', () => {
  const tasks = [];

  const { container } = render(<List tasks={tasks} />);

  expect(container).toHaveTextContent('할 일이 없어요!');
});

test('할일 목록을 확인할 수 있다.', () => {
  const tasks = [{
    id: 1,
    title: '뭐라도 하기',
  },
  {
    id: 2,
    title: '뭐라도 했니',
  },
  {
    id: 3,
    title: '뭐라도 안하니',
  }];

  const { container } = render(<List tasks={tasks} />);

  expect(container).not.toHaveTextContent('할 일이 없어요!');
  tasks.forEach((task) => {
    expect(container).toHaveTextContent(task.title);
  });
  expect(container.firstChild).toMatchSnapshot();
});
