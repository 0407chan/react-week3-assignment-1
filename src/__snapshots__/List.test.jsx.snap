// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`할일 목록을 확인할 수 있다. 1`] = `
<ol>
  <li>
    뭐라도 하기
    <button
      type="button"
    >
      완료
    </button>
  </li>
  <li>
    뭐라도 했니
    <button
      type="button"
    >
      완료
    </button>
  </li>
  <li>
    뭐라도 안하니
    <button
      type="button"
    >
      완료
    </button>
  </li>
</ol>
`;
